src/utils/FishingUtils.ts

```diff
@@ -0,0 +1,106 @@
+export function fishForFHIRBestVersion(
+  fisher: Fishable,
+  item: string,
+  ...types: Type[]
+): any | undefined {
+  let result = fisher.fishForFHIR(item, ...types);
+
+  // If the item was not found, try slicing off a version and searching for any version
+  if (result == null && item != null && item.includes('|')) {
+    const [base, ...versionParts] = item.split('|');
+    const version = versionParts.join('|') || null;
+    result = fisher.fishForFHIR(base, ...types);
+    if (version != null && result?.version != null && version != result?.version) {
+      logger.warn(
+        `The ${base} definition was specified with version ${version}, but SUSHI found version ${result.version}`
+      );
+    }
+  }
+
+  return result;
+}
+
+// Use the provided fisher to fish for the item's metadata.
+// If found, return it. If not found, try to remove any |version info and fish for any available version.
+export function fishForMetadataBestVersion(
+  fisher: Fishable,
+  item: string,
+  ...types: Type[]
+): Metadata | undefined {
+  if (fisher == null) {
+    return;
+  }
+
+  let metadata = fisher.fishForMetadata(item, ...types);
+
+  // If the item was not found, try slicing off a version and searching for any version
+  if (metadata == null && item != null && item.includes('|')) {
+    const [base, ...versionParts] = item.split('|');
+    const version = versionParts.join('|') || null;
+    metadata = fisher.fishForMetadata(base, ...types);
+    if (version != null && metadata?.version != null && version != metadata?.version) {
+      logger.warn(
+        `The ${base} definition was specified with version ${version}, but SUSHI found version ${metadata?.version}`
+      );
+    }
+  }
+
+  return metadata;
+}
+
+// Use the provided tank to fish for the item, which might have a |version appended.
+// If the specified version is found, return it. If not found, try to remove any |version info and fish for any available version.
+export function fishInTankBestVersion(
+  tank: FSHTank,
+  item: string,
+  ...types: Type[]
+):
+  | Profile
+  | Extension
+  | Logical
+  | Resource
+  | FshValueSet
+  | FshCodeSystem
+  | Instance
+  | Invariant
+  | RuleSet
+  | Mapping
+  | undefined {
+  let result = tank.fish(item, ...types);
+
+  // If the item was not found, try slicing off a version and searching for any version
+  if (result == null && item != null && item.includes('|')) {
+    const [base, ...versionParts] = item.split('|');
+    const version = versionParts.join('|') || null;
+    result = tank.fish(base, ...types);
+    if (result != null) {
+      const resultVersion = getVersionFromFshDefinition(result, tank.config.version);
+      if (version != null && resultVersion != null && version != resultVersion) {
+        logger.warn(
+          `The ${base} FSH definition was specified with version ${version}, but SUSHI found version ${resultVersion}`
```

Comment:
And again, make sure the final language gets reflected here.

Reply:
Part of [18b8ba8](https://github.com/FHIR/sushi/pull/1275/commits/18b8ba89e0c6aa3a9ec12bfc0c288154ec6be7f5) (although you must know that by now ;) )